### ISSUE TITLE
grep(pattern) not working properly in Seurat 3.0.1

### ISSUE THREAD

**Issue by cesco-lemon** [satijalab/seurat#1678]
Hi everyone!
I had a quick question, in some of my datasets I've used a function similar to the "percent.mito" too look at other group of genes to help me filter cells, e.g. ribosomal genes "RPS", using the codes below:

mito.genes <- grep(pattern = "^MT-", x = rownames(x = object@data), value = TRUE)
percent.mito <- Matrix::colSums(object@raw.data[mito.genes, ])/Matrix::colSums(object@raw.data)
object <- AddMetaData(object = object, metadata = percent.mito, col.name = "percent.mito")
RPS.genes <- grep(pattern = "^RPS", x = rownames(x = object@data), value = TRUE)
percent.RPS <- Matrix::colSums(object@raw.data[RPS.genes, ])/Matrix::colSums(object@raw.data)
object <- AddMetaData(object = object, metadata = percent.RPS, col.name = "percent.RPS")

This worked very well and helped me identify some outliers in my dataset. Now I'm moving into Seurat v3.0.1 and used:
A[["percent.mito"]] <- PercentageFeatureSet(A, pattern = "^MT-")

Which worked and made my code shorter which always helps. But when I tried to do:

A[["percent.RPS"]] <- PercentageFeatureSet(A, pattern = "^RPS-")

A new column with "percent.RPS" appears in my meta.data which I assume it makes it work but the column is full of zeros. The "percent.mito" column works and it looks the same then when I used to do it with the old v2.3.4 code.

Do you have any idea of what's going wrong? I don't get an error in the code because it does add a column but I have the feeling that it just can't identify genes following the pattern. I've also tried:
A[["percent.RPS"]] <- PercentageFeatureSet(A, pattern = "^RPS")
with no hyphen since the genes actually don't have the hyphen in the name but it still doesn't work.
Any clue?

Thank you!

**Comment by timoast** [comment]
Hi, can you post the output of `grep(pattern = "^RPS", x = rownames(A), value = TRUE)`?

Can you also provide the output of `sessionInfo()`?

**Comment by cesco-lemon** [comment]
If I have Seurat v2.3.4 and I ran:

mito.genes <- grep(pattern = "^MT", x = rownames(x = A@data), value = TRUE)
RPS.genes <- grep(pattern = "^RPS", x = rownames(x = A@data), value = TRUE)

I get similar outputs, which is two tables "mito.genes" and "RPS.genes". Each of them has the mitochondiral and the ribosomal genes that I wanted to grep. So with v2.3.4 it all worked fine for several of my analyses.

The problem is that with Seurat v3.0.1 the lines are different. If I ran this:
A[["percent.mito"]] <- PercentageFeatureSet(A, pattern = "^MT-")
VlnPlot(A, features = c("nFeature_RNA", "nCount_RNA", "percent.mito"), ncol = 3)
Nothing changes between v2 and v3 since I get a similar violinplot.

BUT if I ran:
A[["percent.RPS"]] <- PercentageFeatureSet(A, pattern = "^RPS-")

I don't get any error until I try to run the following:
VlnPlot(A, features = c("nFeature_RNA", "nCount_RNA", "percent.RPS"), nCol = 3)
Error in plot_to_gtable(x) : 
  Argument needs to be of class "ggplot", "gtable", "grob", "recordedplot", or a function that plots to an R graphicsdevice when called, but is a numeric
In addition: Warning message:
In SingleExIPlot(type = type, data = data[, x, drop = FALSE], idents = idents,  :
  All cells have the same value of percent.RPS.


If I look at my metadata in v3, I get this output:
head(A@meta.data)
                 orig.ident nCount_RNA nFeature_RNA percent.mito percent.RPS
GCAAACTAGTACGACG    A      40038         6867    2.1654428           0
AAGGCAGAGTGTTAGA    A      36723         6401    2.9137053           0
CGGGTCATCCAAATGC    A      36617         6526    0.4014529           0
TTGGCAATCTTACCTA    A      36197         6023    0.9945576           0
GAACATCCATGCAACT    A      35687         6073    2.4574775           0
TTAACTCCACTGTGTA    A      34367         6286    1.1028021           0

So in v3 I can definitely grep the "MT-" genes in a similar way as I did in older versions of Seurat but apparently I cannot grep other patterns like "RPS-" since the meta.data are all zeros. I have double-checked in if I go back to v2.3.4 and look at the head(A@meta.data) the "percent.RPS" column has actual values. Unfortunately, I can only run v2.3.4 in my Rstudio but my data is huge and on the cluster I'm using I can only use Seurat v3.0.1.

So how can I adapt the new A[["percent.mito"]] <- PercentageFeatureSet(A, pattern = "^MT-") so that I can look at different patterns?
